File: src/debug-file.cpp
#include <filesystem>
#include "debug-file.h"

DebugFile::DebugFile(str outDir, str srcPath, str options)
: file(nullptr), outDir(outDir), srcPath(srcPath) {
    path = srcPath + "debug";
    path = std::filesystem::path(path).filename();
    path = fmt::format("{}/{}", outDir, path);

    if(in('d', options)) {
        file = new std::ofstream(path);
        write(fmt::format("{}:", srcPath));
    }
}

DebugFile::~DebugFile() {
    if(file) file->close();
    delete file;
}

void DebugFile::write(const str &s, str end) {
    if(file) *file << s << end;
}

File: app/src/main/kotlin/sea/DebugFile.kt
package sea

import java.io.File

data class DebugFile(val outDir: String, val srcPath: String, val options: String) {
    var path: String

    val enabled: Boolean
        get() = "d" in options

    init {
        path = srcPath + "debug"
        path = File(path).name
        path = "$outDir/$path"

        if(enabled) File(path).writeText("$srcPath\n\n")
    }

    fun write(string: String, end: String = "\n\n") {
        if(enabled) File(path).appendText("$string$end")
    }
}
